app/middleware/parameter-validation.ts

```diff
@@ -30,6 +34,46 @@ function validateLinkTypeParameter(req: HarmonyRequest): void {
   }
 }
 
+/**
+ * Validate that the bucket name provided is in the same AWS region as the given region
+ *
+ * @param bucketName - The name of the s3 bucket
+ * @param region - The name of the aws region
+ */
+async function validateBucketIsInRegion(bucketName: string, region: string): Promise<void> {
+  try {
+    const bucketRegion = await defaultObjectStore().getBucketRegion(bucketName);
+    if (bucketRegion != region) {
+      throw new RequestValidationError(`Destination bucket '${bucketName}' must be in the '${region}' region, but was in '${bucketRegion}'.`);
```

Comment:
This message is not exactly actionable. I think we want to write some kind of document first which is out of scope for this ticket, then reference it here.

Reply:
Agreed - the scope of this ticket is for the validation of the region. As part of implementing the ticket that provides full documentation on the destinationUrl issue we should update this documentation - you can throw a TODO comment with the ticket number in here to make tracking it easier.

Reply:
The ticket for writing the cross account s3 delivery setup document is HARMONY-1218.